projects/plugins/jetpack/extensions/global.d.ts

```diff
@@ -71,6 +71,12 @@ declare global {
 			siteLocale: string;
 			'ai-assistant': SiteAIAssistantFeatureEndpointResponseProps;
 			screenBase?: string;
+			search?: {
+				instant_search_enabled: boolean;
+			};
+			jetpack: {
+				is_current_user_connected: boolean;
```

Comment:
This is already available in the JP_CONNECTION_INITIAL_STATE object https://github.com/Automattic/jetpack/pull/33064/files#diff-5ed459628f52ee22070733dd3841879ef252227b0532a586d545267cacfae371R34

Reply:
Can we repurpose these functions in the ai-assistant lib? https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/jetpack/extensions/blocks/ai-assistant/lib/connection/index.ts

Reply:
> This is already available in the JP_CONNECTION_INITIAL_STATE object

Yes, and it is already available on `Jetpack_Editor_Initial_State` object as well.
I decided to use the Jetpack_Editor_Initial_State because it's consistently available on both wpcom and Jetpack.

I only added a type to a field that already exists.

> Can we repurpose these functions in the ai-assistant lib?

I was considering it, but it's not the same feature/package. I can imagine them not being available at the same time.
Ai Chat is not a part of jetpack ai, but jetpack search.

Reply:
> Ai Chat is not a part of jetpack ai, but jetpack search.

Technically both can be considered part of Jetpack (the plugin) right now. Since, for example, the block is not available in the Jetpack Search plugin until it's isolated into a consumable package.

Reply:
Pushed a change that uses `Automattic\Jetpack\Search\Initial_State as Search_Initial_State` and passing it within it's own initial state object as suggested.